src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -77,28 +81,44 @@
 	default Optional<T> getOptionalSingle(Event e) {
 		return Optional.ofNullable(getSingle(e));
 	}
-	
+
 	/**
 	 * Get all the values of this expression. The returned array is empty if this expression doesn't have any values for the given event.
 	 * <p>
 	 * The returned array must not contain any null values.
 	 * <p>
 	 * Do not use this in conditions, use {@link #check(Event, Checker, boolean)} instead.
 	 * 
-	 * @param e The event
+	 * @param event The event the expression is being used in.
 	 * @return An array of values of this expression which must neither be null nor contain nulls, and which must not be an internal array.
 	 */
-	public T[] getArray(final Event e);
-	
+	public T[] getArray(Event event);
+
+	/**
+	 * Get all the values of this expression and then convert them using the provided getter for easy of access.
+	 * <p>
+	 * This method will use {@link Expression#getArray(Event)} to use non null values in the Converter.
+	 * 
+	 * @param <G> The returning type of the converter.
+	 * @param event The event the expression is being used in.
+	 * @param converter The converter to do the convertering to the type.
+	 * @return The values completed by the converter.
```

Comment:
It should be made clear that the returned array values will not be null, even if the converter returned null values, they're filtered out:
https://github.com/SkriptLang/Skript/blob/8dc04919b07ee9a7fd4a1c0afca148edd030dcf6/src/main/java/ch/njol/skript/registrations/Converters.java#L390-L392